src/main/java/com/theelixrlabs/healthcare/controller/DoctorController.java

```diff
@@ -33,4 +38,22 @@ public ResponseEntity<SuccessResponse<DoctorDto>> createDoctor(@Valid @RequestBo
         DoctorDto createdDoctor = doctorService.saveDoctor(doctorDto);
         return new ResponseEntity<>(new SuccessResponse<>(true, createdDoctor), HttpStatus.OK);
     }
+
+    /**
+     * Retrieves a doctor by their UUID.
+     *
+     * @param id The ID of the doctor as a String, which will be converted to a UUID.
+     * @return ResponseEntity containing a success response with the retrieved DoctorDto.
+     */
+    @GetMapping(DoctorConstants.GET_DOCTOR_BY_ID_ENDPOINT)
+    public ResponseEntity<?> getDoctorById(@PathVariable("doctorId") String id) {
```

Comment:
String "doctorId" should be moved to DoctorConstants

Reply:
Resolved it ..